src/Orleans/Runtime/RequestContext.cs

```diff
@@ -42,6 +41,11 @@ public static class RequestContext
         internal const string ORLEANS_REQUEST_CONTEXT_KEY = "#ORL_RC";
         internal const string PING_APPLICATION_HEADER = "Ping";
 
+#if NETSTANDARD_TODO
+        private static readonly AsyncLocal<Guid> ActivityId = new AsyncLocal<Guid>();
+        private static readonly AsyncLocal<Dictionary<string, object>> CallContextData = new AsyncLocal<Dictionary<string, object>>();
```

Comment:
I'm not entirely convinced with how the ActivityId is being handled here. If I understand correctly, the only way of setting it is by using `E2_E_TRACING_ACTIVITY_ID_HEADER` in the contextData. If that's the case, why is it being extracted to its own AsyncLocal value? A little clarification on how this is expected to work would be very valuable.